client/signup/config/step-components.js

```diff
@@ -2,70 +2,71 @@
-	'site-topic': loadStep( 'site-topic' ),
-	'site-type': loadStep( 'site-type' ),
-	survey: loadStep( 'survey' ),
-	'survey-user': loadStep( 'survey-user' ),
-	test: loadStep( 'test-step' ),
-	themes: loadStep( 'theme-selection' ),
-	'website-themes': loadStep( 'theme-selection' ),
-	'blog-themes': loadStep( 'theme-selection' ),
-	'themes-site-selected': loadStep( 'theme-selection' ),
-	user: loadStep( 'user' ),
-	'oauth2-user': loadStep( 'user' ),
-	'oauth2-name': loadStep( 'user' ),
-	displayname: loadStep( 'user' ),
-	'reader-landing': loadStep( 'reader-landing' ),
-	// Steps with preview
-	'site-style-with-preview': loadStep( 'site-style' ),
-	'site-topic-with-preview': loadStep( 'site-topic' ),
-	'domains-with-preview': loadStep( 'domains' ),
-	'site-title-with-preview': loadStep( 'site-title' ),
+const stepNameToFilenameMap = {
+	about: 'about',
+	'clone-start': 'clone-start',
+	'clone-destination': 'clone-destination',
+	'clone-credentials': 'clone-credentials',
+	'clone-point': 'clone-point',
+	'clone-jetpack': 'clone-jetpack',
+	'clone-ready': 'clone-ready',
+	'clone-cloning': 'clone-cloning',
+	'creds-confirm': 'creds-confirm',
+	'creds-complete': 'creds-complete',
+	'creds-permission': 'creds-permission',
+	domains: 'domains',
+	'domains-store': 'domains',
+	'domain-only': 'domains',
+	'domains-theme-preselected': 'domains',
+	'domains-launch': 'domains',
+	'from-url': 'import-url',
+	launch: 'launch-site',
+	plans: 'plans',
+	'plans-launch': 'plans',
+	'plans-personal': 'plans',
+	'plans-premium': 'plans',
+	'plans-business': 'plans',
+	'plans-store-nux': 'plans-atomic-store',
+	'plans-site-selected': 'plans-without-free',
+	site: 'site',
+	'rebrand-cities-welcome': 'rebrand-cities-welcome',
+	'rewind-migrate': 'rewind-migrate',
+	'rewind-were-backing': 'rewind-were-backing',
+	'rewind-add-creds': 'rewind-add-creds',
+	'rewind-form-creds': 'rewind-form-creds',
+	'site-or-domain': 'site-or-domain',
+	'site-picker': 'site-picker',
+	'site-style': 'site-style',
+	'site-title': 'site-title',
+	'site-title-without-domains': 'site-title',
+	'site-topic': 'site-topic',
+	'site-type': 'site-type',
+	survey: 'survey',
+	'survey-user': 'survey-user',
+	test: 'test-step',
+	themes: 'theme-selection',
+	'website-themes': 'theme-selection',
+	'blog-themes': 'theme-selection',
+	'themes-site-selected': 'theme-selection',
+	user: 'user',
+	'oauth2-user': 'user',
+	'oauth2-name': 'user',
+	displayname: 'user',
+	'reader-landing': 'reader-landing',
+	'site-style-with-preview': 'site-style',
+	'site-topic-with-preview': 'site-topic',
+	'domains-with-preview': 'domains',
+	'site-title-with-preview': 'site-title',
 };
+
+export const asyncLoadStep = stepName =>
+	import(
+		/* webpackChunkName: "async-load-signup-steps-[request]", webpackInclude: /signup\/steps\/[a-z-]+\/index.jsx$/ */ `signup/steps/${ stepName }`
+	);
+
+const loadStepComponent = memoize( stepName => React.lazy( () => asyncLoadStep( stepName ) ) );
```

Comment:
There's a bug here: the module loaded shouldn't be `stepName`, but `stepNameToFilenameMap[ stepName ]`. After fixing, the `getStepComponent` and `loadStepComponent` functions should be the same.